packages/validator/contracts/ERC20Executor.sol

```diff
@@ -0,0 +1,72 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.10;
+
+import "./SingleClaimantExecutor.sol";
+import "@openzeppelin/contracts/contracts/token/ERC20/IERC20.sol";
+
+/**
+ * @dev An abstract implementation of an Executor that only allows claims that have a higher nonce than previously seen to execute.
+ *      To use, subclass and override `executeClaim` and `metadata`, being sure to call `super` inside `executeClaim` before doing anything else.
+ *      This executor expects the first 32 bytes of `claimData` to be the nonce; you may optionally use extra data for your own purposes.
+ */
+abstract contract ERC20Executor is SingleClaimantExecutor {
+    event ClaimedERC20(address issuer, address from, address beneficiary, address token, uint256 amount, uint256 expiration);
+    
+    error ClaimCodeExpired();
+
+    constructor(address _validator) SingleClaimantExecutor(_validator) { }
+  
+    /**
+     * @dev Executes a claim that has been verified by the `ValidatorRegistry`. Implementers must check that this function
+     *      was called by a registry they recognise, and that any conditions in claimData such as replay protection are met
+     *      before acting on the request.
+     * @param issuer The account that issued the claim.
+     * @param claimant The account that is entitled to make the claim.
+     * @param beneficiary The account that the claim should benefit.
+     * @param claimData Claim data provided by the issuer.
+     * @param executorData Contextual information stored on the ValidatorRegistry for this issuer.
+     */
+    function executeClaim(address issuer, address claimant, address beneficiary, bytes calldata claimData, bytes calldata executorData) public virtual override {
+        super.executeClaim(issuer, claimant, beneficiary, claimData, executorData);
```

Comment:
This is going to need some kind of access control. Otherwise, anyone could create their own issuer that uses this as an executor, and send themselves all the tokens!

Reply:
Added `onlyIssuer` modifier to BaseExecutor and the modifier is added to ERC20Executor#executeClaim for access control